### ISSUE TITLE
RF showing on blocks with no storage for it

### ISSUE THREAD

**Issue by LemADEC** [McJty/TheOneProbe#274]
As of 1.12-1.4.25, TOP will shows "0 RF" for blocks that actually don't store any RF.
Consider calling IEnergyHandler.getMaxEnergyStored() and check value is > 0 before displaying the RF line.

**Comment by josephcsible** [comment]
I tried a few vanilla blocks (which obviously don't have or use RF), and this didn't happen. Can you give an example of a block where it does happen?

**Comment by LemADEC** [comment]
It happens on block who have the API but still don't store energy in them.

**Comment by josephcsible** [comment]
I think I'd consider that a bug in the mod adding such block. Again, can you give a specific example of a block that does that?

**Comment by LemADEC** [comment]
The RedstoneFlux API allows you to have the interface implemented without any energy stored in it. 
This is needed in my WarpDrive mod due to class inheritance and affect a couple blocks (Ship controllers and ForceField relays notably).

**Comment by McJty** [comment]
While this is indeed possible it leaves the question if this is actually a good idea to do. Implementing this interface will also imply that cables and such will connect to these blocks while this is not useful at all. Not sure if that's a very good idea to be honest

**Comment by LemADEC** [comment]
Cables won't connect because the faces are disabled for sending and receiving energy.

**Comment by josephcsible** [comment]
I agree with @McJty here. This seems like a flaw in your class hierarchy. The point of having the three separate interfaces (IEnergyHandler, IEnergySink, and IEnergySource) is that things only implement the ones that apply to them. You've chosen to ignore this system, and are asking us to add a hack to replicate its functionality.

**Comment by LemADEC** [comment]
![image](https://user-images.githubusercontent.com/4776032/50425375-194a3400-0875-11e9-977b-268e7673dc2c.png)

**Comment by josephcsible** [comment]
Okay, but why can't you just change your class hierarchy so that IEnergyHandler isn't implemented by one of your base classes?

**Comment by McJty** [comment]
> Cables won't connect because the faces are disabled for sending and receiving energy.

That will not work for mods like (for example) XNet since they also connect if the block supports any kind of energy

**Comment by LemADEC** [comment]
There's always different ways to implement the same feature. In the present case, the existing inheritance removes lots of code duplication.

A better question is why would re-implement mod A because mod B doesn't want to follow their common API?

**Comment by LemADEC** [comment]
Sounds like XNet should check the sided connection like ThermalDynamics cables do.

**Comment by McJty** [comment]
XNet connectors can check from any side (even a side different from the one they are physically connected too). That's one of the features of XNet. So when they check if a cable can connect to a block I don't have a side to use so I pass null

**Comment by LemADEC** [comment]
XNet don't connect whether energy is enabled or not:
![image](https://user-images.githubusercontent.com/4776032/50425485-86f75f80-0877-11e9-8b05-835de4a18d6e.png)
Assuming that's how we're supposed to use those cables.

**Comment by McJty** [comment]
No, that's not a connector. Cables don't connect to anything. It is connectors that make connections

**Comment by josephcsible** [comment]
> A better question is why would re-implement mod A because mod B doesn't want to follow their common API?

Your mod is the one that doesn't want to follow the common API, not ours.

**Comment by LemADEC** [comment]
Xnet connector does show as connected to both. The controller does return false when given a null facing, so I guess Xnet doesn't check more than the interface implementation?
